cpp/src/arrow/compute/kernels/scalar_arithmetic_test.cc

```diff
@@ -1333,6 +1400,305 @@ TYPED_TEST(TestUnaryArithmeticFloating, AbsoluteValue) {
   }
 }
 
+constexpr std::initializer_list<RoundMode> kRoundModes = {
```

Comment:
Having a list like this is fine. We probably don't want to expose EnumTraits<Foo> (unless you really want to make another internal header/translation unit to hold the definitions).